huaweicloud/services/cts/resource_huaweicloud_cts_notification.go

```diff
@@ -129,12 +149,51 @@ func buildNotificationCreateRequestBody(d *schema.ResourceData) *cts.CreateNotif
 		Operations:       buildKeyOperationOpts(d),
 		NotifyUserList:   buildNotifyUserOpts(d),
 		TopicId:          utils.String(d.Get("smn_topic").(string)),
+		Filter:           buildKeyFilterOpts(d),
 	}
 
 	log.Printf("[DEBUG] creating CTS key events notification options: %#v", reqBody)
 	return &reqBody
 }
 
+func buildKeyFilterOpts(d *schema.ResourceData) *cts.Filter {
+	rawFilter := d.Get("filter").([]interface{})
+	if len(rawFilter) == 0 {
+		return nil
+	}
+
+	filter := cts.Filter{}
+	for _, filterItem := range rawFilter {
+		filterData := filterItem.(map[string]interface{})
+		filter.Rule = make([]string, 0)
+
+		if ruleList, ok := filterData["rule"].([]interface{}); ok {
+			for _, rule := range ruleList {
+				filter.Rule = append(filter.Rule, rule.(string))
+			}
+		}
+
+		conditionStr := filterData["condition"].(string)
+		switch conditionStr {
```

Comment:
```suggestion
		switch filterData["condition"].(string) {
```
this variable is used only once

Reply:
updated.